src/lib/core/CHIPTLVReader.cpp

```diff
@@ -128,83 +128,112 @@ CHIP_ERROR TLVReader::Get(bool & v)
 CHIP_ERROR TLVReader::Get(int8_t & v)
 {
     uint64_t v64   = 0;
-    CHIP_ERROR err = Get(v64);
+    CHIP_ERROR err = Get(v64, sizeof(int8_t));
     v              = CastToSigned(static_cast<uint8_t>(v64));
     return err;
 }
 
 CHIP_ERROR TLVReader::Get(int16_t & v)
 {
     uint64_t v64   = 0;
-    CHIP_ERROR err = Get(v64);
+    CHIP_ERROR err = Get(v64, sizeof(int16_t));
     v              = CastToSigned(static_cast<uint16_t>(v64));
     return err;
 }
 
 CHIP_ERROR TLVReader::Get(int32_t & v)
 {
     uint64_t v64   = 0;
-    CHIP_ERROR err = Get(v64);
+    CHIP_ERROR err = Get(v64, sizeof(int32_t));
     v              = CastToSigned(static_cast<uint32_t>(v64));
     return err;
 }
 
 CHIP_ERROR TLVReader::Get(int64_t & v)
 {
     uint64_t v64   = 0;
-    CHIP_ERROR err = Get(v64);
+    CHIP_ERROR err = Get(v64, sizeof(int64_t));
     v              = CastToSigned(v64);
     return err;
 }
 
 CHIP_ERROR TLVReader::Get(uint8_t & v)
 {
     uint64_t v64   = 0;
-    CHIP_ERROR err = Get(v64);
+    CHIP_ERROR err = Get(v64, sizeof(uint8_t));
     v              = static_cast<uint8_t>(v64);
     return err;
 }
 
 CHIP_ERROR TLVReader::Get(uint16_t & v)
 {
     uint64_t v64   = 0;
-    CHIP_ERROR err = Get(v64);
+    CHIP_ERROR err = Get(v64, sizeof(uint16_t));
     v              = static_cast<uint16_t>(v64);
     return err;
 }
 
 CHIP_ERROR TLVReader::Get(uint32_t & v)
 {
     uint64_t v64   = 0;
-    CHIP_ERROR err = Get(v64);
+    CHIP_ERROR err = Get(v64, sizeof(uint32_t));
     v              = static_cast<uint32_t>(v64);
     return err;
 }
 
 CHIP_ERROR TLVReader::Get(uint64_t & v)
 {
+    uint64_t v64   = 0;
+    CHIP_ERROR err = Get(v64, sizeof(uint64_t));
+    v              = v64;
+    return err;
+}
+
+static inline CHIP_ERROR VerifySize(size_t requestedSize, size_t actualSize)
```

Comment:
nit: consider not having this extra indirection.

I would find a `ReturnErrorCodeIf(size < sizeof(int16_t), CHIP_ERROR_WRONG_TLV_TYPE)` easier to understand right away and it would also avoid a temporary chip_error local variable.